app/models/rbac.rb

```diff
@@ -248,7 +248,19 @@ def self.find_targets_with_user_group_rbac(klass, scope, rbac_filters, find_opti
     end
   end
 
+  def self.find_options_for_tenant(klass, user_or_group, find_options = {})
+    tenant_id = user_or_group.try(:current_tenant, :id)
+    return find_options unless tenant_id
+
+    tenant_id_clause = {klass.table_name => {:tenant_id => [tenant_id, nil]}}
+    find_options[:conditions] = MiqExpression.merge_where_clauses(find_options[:conditions], tenant_id_clause)
+    find_options
+  end
+
   def self.find_targets_with_rbac(klass, scope, rbac_filters, find_options = {}, user_or_group = nil)
+    # TODO: check if these find_options should be duplicated/modified in place
```

Comment:
in the past we have been avoiding this.
since we tend to call this with a merge, that allows us to avoid actually manipulating.

but it is probably no harm